### PR TITLE
ISPN-6799 OOB threads block sending remote get responses

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-6799
https://issues.jboss.org/browse/ISPN-6849

* Set the NO_FC flag on RPC response messages
* Upgrade to JGroups 3.6.10.Final

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -72,7 +72,7 @@
       <version.cdi>1.2</version.cdi>
       <version.jboss.marshalling>1.4.11.Final</version.jboss.marshalling>
       <version.jboss.logging>3.3.0.Final</version.jboss.logging>
-      <version.jgroups>3.6.9.Final</version.jgroups>
+      <version.jgroups>3.6.10.Final</version.jgroups>
```
Comment (pruivo) [inline bom/pom.xml]: shouldn't it be in a separate JIRA/PR?

File: core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java [hunk 2]
```diff
@@ -240,11 +242,30 @@ public String toString() {
...
+               rsp_buf = rsp_marshaller.objectToBuffer(t);
+               is_exception = true;
+            } catch (Throwable tt) {
+               log.errorMarshallingObject(tt, retVal);
+               return;
+            }
+         }
+
+         // Always set the NO_FC flag
+         short flags = (short) (req.getFlags() | NO_FC.value() & ~Message.Flag.RSVP.value() & ~Message.Flag.SCOPED.value());
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java]: I would remove the ```INTERNAL``` flag and set ```NO_TOTAL_ORDER``` flag (I don't see a case in ISPN where we need to send the reply in TO :) )

File: core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java [hunk 3]
```diff
@@ -44,6 +44,7 @@
 import java.util.concurrent.TimeUnit;
 
 import static org.infinispan.remoting.transport.jgroups.JGroupsTransport.fromJGroupsAddress;
+import static org.jgroups.Message.Flag.NO_FC;
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java]: BTW, there is an unused import of ```IOException``` that can be removed.

File: core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java (hunk 2 above)
Comment (danberindei) [inline core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java]: Will do.

**Comment by danberindei** [comment]
Updated, with a separate JIRA for the JGroups update.

**Comment by pruivo** [comment]
compile error
```
[ERROR] /home/pedro/private-work/write-infinispan/server/integration/jgroups/src/main/java/org/infinispan/server/jgroups/subsystem/JGroupsExtension.java:[55,46] cannot find symbol
[ERROR] symbol:   variable CUSTOM_LOG_FACTORY
[ERROR] location: class org.jgroups.Global
[ERROR] /home/pedro/private-work/write-infinispan/server/integration/jgroups/src/main/java/org/infinispan/server/jgroups/subsystem/JGroupsExtension.java:[56,46] cannot find symbol
[ERROR] symbol:   variable CUSTOM_LOG_FACTORY
[ERROR] location: class org.jgroups.Global
[ERROR] -> [Help 1]
```

**Comment by danberindei** [comment]
@pruivo Updated. Weird that I didn't notice the compilation failure, especially since I had already fixed it once :)

**Comment by pruivo** [comment]
integrated! thanks @danberindei!

**Pull request closed without merging**